### PR TITLE
cli: integrate craft-cli and add version command (CRAFT-730)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Handle message output and command line arguments parsing using
craft-cli. Implement the `version` command and fall back to the
legacy snapcraft implementation to handle all other functionality.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/__init__.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Publish your app for Linux users for desktop, cloud, and IoT."""
+
+import os
+
+import pkg_resources  # type: ignore
```
Comment (sergiusens) [inline snapcraft/__init__.py:21]: why do we need to ignore type here?

File: snapcraft/cli.py [hunk 2]
```diff
@@ -16,9 +16,45 @@
...
+import craft_cli
+from craft_cli import EmitterMode, emit
+
+from snapcraft import __version__
 from snapcraft_legacy.cli import legacy
 
+from . import commands
+
+COMMAND_GROUPS = [
+    craft_cli.CommandGroup("Basic", [commands.VersionCommand]),
```
Comment (sergiusens) [inline snapcraft/cli.py:30]: The snap command has this under "Other"

File: tests/unit/conftest.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+import os
+import tempfile
+from pathlib import Path
+
+import pytest
+from craft_cli import messages
+
+
+class RecordingEmitter:
```
Comment (sergiusens) [inline tests/unit/conftest.py:25]: I assume this is copied over? Maybe add a TODO to use a common package?

**Review by sergiusens — APPROVED** [review]
Looks good, have a couple of non binding comments and questions

File: tests/unit/conftest.py (hunk 3 above)
Comment (cmatsuoka) [reply tests/unit/conftest.py:25]: This will be removed once testing fixtures are provided by craft-cli itself, or by pytest-craft-cli.

**Comment by cmatsuoka** [comment]
There's an additional issue here: we should leave `--help` to be handled by the legacy implementation until the new help is complete with all valid commands, otherwise it only lists what's implemented (currently only `version` and `--version`).

File: snapcraft/cli.py (hunk 2 above)
Comment (cmatsuoka) [reply snapcraft/cli.py:30]: Let's move it to Other then.

File: snapcraft/__init__.py (hunk 1 above)
Comment (cmatsuoka) [reply snapcraft/__init__.py:21]: Good catch, this is not needed when `types-setuptools` is installed.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**